sdk/communication/azure-communication-common/src/main/java/com/azure/communication/common/CommunicationTokenRefreshOptions.java

```diff
@@ -2,66 +2,89 @@
-     *                           tokenRefresher before token expiry by minutes set
-     *                           with setCallbackOffsetMinutes or default value of
-     *                           two minutes
+     * @param tokenRefresher The callback function that acquires a fresh token from the Communication Identity API, e.g. by calling the CommunicationIdentityClient
+     * @param refreshProactively Determines whether the token should be proactively renewed prior to its expiry or on demand.
      */
     public CommunicationTokenRefreshOptions(Supplier<Mono<String>> tokenRefresher, boolean refreshProactively) {
-        this.tokenRefresher = tokenRefresher;
-        this.refreshProactively = refreshProactively;
-        this.initialToken = null;
+        this(tokenRefresher, refreshProactively, null, getDefaultRefreshTimeBeforeTokenExpiry());
     }
 
-     /**
+    /**
      * Creates a CommunicationTokenRefreshOptions object
      *
-     * @param tokenRefresher the token refresher to provide capacity to fetch fresh token
-     * @param refreshProactively when set to true, turn on proactive fetching to call
-     *                           tokenRefresher before token expiry by minutes set
-     *                           with setCallbackOffsetMinutes or default value of
-     *                           two minutes
-     * @param initialToken the optional serialized JWT token
+     * @param tokenRefresher The callback function that acquires a fresh token from the Communication Identity API, e.g. by calling the CommunicationIdentityClient
+     * @param refreshProactively Determines whether the token should be proactively renewed prior to its expiry or on demand.
+     * @param initialToken The optional serialized JWT token
      */
     public CommunicationTokenRefreshOptions(Supplier<Mono<String>> tokenRefresher, boolean refreshProactively, String initialToken) {
+        this(tokenRefresher, refreshProactively, initialToken, getDefaultRefreshTimeBeforeTokenExpiry());
+    }
+
+    /**
+     * Creates a CommunicationTokenRefreshOptions object
+     *
+     * @param tokenRefresher The callback function that acquires a fresh token from the Communication Identity API, e.g. by calling the CommunicationIdentityClient
+     * @param refreshProactively Determines whether the token should be proactively renewed prior to its expiry or on demand.
+     * @param initialToken The optional serialized JWT token
+     * @param refreshTimeBeforeTokenExpiry The time span before token expiry that causes the tokenRefresher to be called if refreshProactively is true. For example, setting it to 5 minutes means that 5 minutes before the cached token expires, the proactive refresh will request a new token. The default value is 10 minutes.
+     */
+    public CommunicationTokenRefreshOptions(Supplier<Mono<String>> tokenRefresher, boolean refreshProactively, String initialToken, Duration refreshTimeBeforeTokenExpiry) {
```

Comment:
we need to have an option to configure the options bag without specifying the initialToken...there are two ways to do it:
- add a set method
- add a new constructor `CommunicationTokenRefreshOptions(Supplier<Mono<String>> tokenRefresher, boolean refreshProactively, Duration refreshTimeBeforeTokenExpiry)`

Add a question to the API view, tagging (I think) Jonathan Giles and check what's the best approach and implement it.